bot/exts/utilities/challenges.py

```diff
@@ -0,0 +1,323 @@
+# These colors are representative of the colors that each 'kyu' level represents on codewars.com
+MAPPING_OF_KYU = {
+    8: (221, 219, 218), 7: (221, 219, 218), 6: (236, 182, 19), 5: (236, 182, 19),
+    4: (60, 126, 187), 3: (60, 126, 187), 2: (134, 108, 199), 1: (134, 108, 199),
+}
+
+# Supported languages for a kata on codewars.com
+SUPPORTED_LANGUAGES = {
+    "stable": [
+        "c", "c#", "c++", "clojure", "coffeescript", "coq", "crystal", "dart", "elixir",
+        "f#", "go", "groovy", "haskell", "java", "javascript", "kotlin", "lean", "lua", "nasm",
+        "php", "python", "racket", "ruby", "rust", "scala", "shell", "sql", "swift", "typescript",
+    ],
+    "beta": [
+        "agda", "bf", "cfml", "cobol", "commonlisp", "elm", "erlang", "factor",
+        "forth", "fortran", "haxe", "idris", "julia", "nim", "objective-c", "ocaml",
+        "pascal", "perl", "powershell", "prolog", "purescript", "r", "raku", "reason", "solidity", "vb.net",
+    ]
+}
+
+
+class InformationDropdown(ui.Select):
+    """A dropdown inheriting from ui.Select that allows finding out other information about the kata."""
+
+    def __init__(
+            self,
+            language_embed: Embed,
+            tags_embed: Embed,
+            other_info_embed: Embed,
+            main_embed: Embed,
+    ):
+        options = [
+            SelectOption(
+                label="Main Information",
+                description="See the kata's difficulty, description, etc.",
+                emoji='🌎',
+            ),
+            SelectOption(
+                label='Languages',
+                description='See what languages this kata supports!',
+                emoji=Emojis.reddit_post_text,
+            ),
+            SelectOption(
+                label='Tags',
+                description='See what categories this kata falls under!',
+                emoji=Emojis.stackoverflow_tag,
+            ),
+            SelectOption(
+                label='Other Information',
+                description='See how other people performed on this kata and more!',
+                emoji='ℹ',
+            ),
+        ]
+
+        # We map the option label to the embed instance so that it can be easily looked up later in O(1)
+        self.mapping_of_embeds = {
+            "Main Information": main_embed,
+            "Languages": language_embed,
+            "Tags": tags_embed,
+            "Other Information": other_info_embed,
+        }
+
+        super().__init__(
+            placeholder='See more information regarding this kata',
+            min_values=1,
+            max_values=1,
+            options=options,
+        )
+
+    async def callback(self, interaction: Interaction) -> None:
+        """Callback for when someone clicks on a dropdown."""
+        # Edit the message to the embed selected in the option
+        result_embed = self.mapping_of_embeds[self.values[0]]
+        await self.original_message.edit(embed=result_embed)
```

Comment:
Should probably comment that `original_message` will be set after being initialized.

Reply:
Will do